### PR TITLE
feat(watchmovieshd): add presence

### PR DESCRIPTION
**Pull request by Aarondevp**
## Description 

adding presence for watchmovieshd.ru website

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/M/Musixmatch/dist/metadata.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
 		"curators.musixmatch.com"
 	],
 	"version": "2.0.1",
+>>>>>>> edd2595e328a8c1f8c69a4b31f15b48ec10772c7
```
Comment (theusaf) [inline websites/M/Musixmatch/dist/metadata.json:24]: You should undo the changes to this file and `yarn.lock` and re-push. The PR should only contain changes to your 4 files in `websites/W/watchmovieshd`

File: websites/W/watchmovieshd/presence.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	"iFrameData",
+	(data: { currTime: number; duration: number; paused: boolean }) => {
+		iFrameData = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp,
+			largeImageKey: "logo",
```
Comment (theusaf) [inline websites/W/watchmovieshd/presence.ts:22]: I've been told that Imgur links are preferred for these images for easier maintenance by other developers

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/W/watchmovieshd/presence.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		presenceData.details = "Checking Watchlist";
+	else {
+		const genre = document.querySelector<HTMLHeadingElement>("section.bl h1");
+		if (genre) {
+			presenceData.details = genre.textContent;
+			presenceData.smallImageKey = "search";
+		}
+	}
+
+	presence.setActivity(presenceData);
```
Comment (theusaf) [inline websites/W/watchmovieshd/presence.ts:99]:
```suggestion
	if (presenceData.details) presence.setActivity(presenceData);
	else presence.setActivity();
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — CHANGES_REQUESTED** [review]
Also, ensure the title of this PR follows the [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md).

File: websites/W/watchmovieshd/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+		"name": "Toxicity",
+		"id": "794558195921846292"
+	},
+	"service": "watchmovieshd",
+	"description": {
+		"en": "This is a website where you can watch tv shows and movies for free"
+	},
+	"url": "watchmovieshd.ru",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/2VmNNzq.png",
```
Comment (theusaf) [inline websites/W/watchmovieshd/dist/metadata.json:13]: Can you find/create a higher-resolution logo?
> The image is **required** to have a minimum resolution of `512x512` pixels. You can upsize it using a tool like [waifu2x](http://waifu2x.udp.jp/).
> — Presence Guidelines

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by RisingSunLight42** [comment]
Resolve check

File: websites/W/watchmovieshd/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Toxicity",
+		"id": "794558195921846292"
+	},
+	"service": "watchmovieshd",
+	"description": {
+		"en": "This is a website where you can watch tv shows and movies for free"
```
Comment (RisingSunLight42) [inline websites/W/watchmovieshd/dist/metadata.json:9]: Bad formulation, check if the website has a `description` meta tag.

File: websites/W/watchmovieshd/presence.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+			startTimestamp: browsingTimestamp,
+			largeImageKey: "logo",
+		},
+		{ pathname, href } = document.location,
+		[buttons, image] = await Promise.all([
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("image"),
+		]);
+
+	if (pathname === "/home") presenceData.details = "Browsing";
```
Comment (RisingSunLight42) [inline websites/W/watchmovieshd/presence.ts:30]: I advise you to change to a switch for a better readability since you're using startsWith func every time.

File: websites/W/watchmovieshd/presence.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+		if (title) presenceData.details = title.textContent;
+		if (season) {
+			presenceData.state = season.textContent.split("-")[0].trim();
+			if (episode) presenceData.state += ` - ${episode.textContent.trim()}`;
+		}
+		if (image) {
+			presenceData.largeImageKey = document
+				.querySelector("meta[property='og:image']")
+				.getAttribute("content");
+		}
```
Comment (RisingSunLight42) [inline websites/W/watchmovieshd/presence.ts:46]:
```suggestion
		presenceData.largeImageKey = document
			.querySelector("meta[property='og:image']")
			.getAttribute("content");
```

File: websites/W/watchmovieshd/presence.ts [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+			presenceData.smallImageKey = "play";
+		} else presenceData.smallImageKey = "pause";
+		if (buttons) {
+			presenceData.buttons = [
+				{
+					label: "Watch Series",
+					url: href,
+				},
+			];
+		}
```
Comment (RisingSunLight42) [inline websites/W/watchmovieshd/presence.ts:61]:
```suggestion
		presenceData.buttons = [
			{
				label: "Watch Series",
				url: href,
			},
		];
```

File: websites/W/watchmovieshd/presence.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+	} else if (pathname.startsWith("/movie/")) {
+		const title = document.querySelector<HTMLHeadingElement>(
+			"#watch > div.container > div.watch-extra > div.bl-1 > section.info > div.info > h1"
+		);
+		if (title) presenceData.details = title.textContent;
+		if (image) {
+			presenceData.largeImageKey = document
+				.querySelector("meta[property='og:image']")
+				.getAttribute("content");
+		}
```
Comment (RisingSunLight42) [inline websites/W/watchmovieshd/presence.ts:71]:
```suggestion
		presenceData.largeImageKey = document
			.querySelector("meta[property='og:image']")
			.getAttribute("content");
```

File: websites/W/watchmovieshd/presence.ts [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+			presenceData.smallImageKey = "play";
+		} else presenceData.smallImageKey = "pause";
+		if (buttons) {
+			presenceData.buttons = [
+				{
+					label: "Watch Movie",
+					url: href,
+				},
+			];
+		}
```
Comment (RisingSunLight42) [inline websites/W/watchmovieshd/presence.ts:86]:
```suggestion
		presenceData.buttons = [
			{
				label: "Watch Movie",
				url: href,
			},
		];
```

File: websites/W/watchmovieshd/presence.ts [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+	else if (pathname === "/user/favourite")
+		presenceData.details = "Checking Watchlist";
+	else {
+		const genre = document.querySelector<HTMLHeadingElement>("section.bl h1");
+		if (genre) {
+			presenceData.details = genre.textContent;
+			presenceData.smallImageKey = "search";
+		}
+	}
+
```
Comment (RisingSunLight42) [inline websites/W/watchmovieshd/presence.ts:98]:
```suggestion
if (!buttons) delete presenceData.buttons
if (!image) delete presenceData.largeImageKey
```

**Review by RisingSunLight42 — CHANGES_REQUESTED** [review]
I you need help, don't hesitate to ask.

**Comment by EncryptedDev** [comment]
PR title doesn’t follow repo conventions shown in the requirements

File: websites/M/Musixmatch/dist/metadata.json (hunk 1 above)
Comment (theusaf) [inline websites/M/Musixmatch/dist/metadata.json:24]:
```suggestion
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/W/watchmovieshd/dist/metadata.json (hunk 5 above)
Comment (DarkVillager) [inline websites/W/watchmovieshd/dist/metadata.json:9]: Tried to make something of the meta description (watch movies hd online, watch movies hd free online, free watch movies hd, free watch movies hd online, update daily, hd quality)
```suggestion
		"en": "Watch movies that get updated daily online in hd quality."
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
what the ..... are u doing?

**Comment by RisingSunLight42** [comment]
> what the ..... are u doing?

Trying to remove other presences files.

File: websites/W/watchmovieshd/presence.ts [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+			presenceData.smallImageKey = "play";
+		} else presenceData.smallImageKey = "pause";
+		presenceData.buttons = [
+			{
+				label: "Watch Movie",
+				url: href,
+			},
+		];
+	} else if (pathname === "/user/profile")
+		presenceData.details = "Checking Profile";
```
Comment (EncryptedDev) [inline websites/W/watchmovieshd/presence.ts:80]: The verb choice here is poor. `Viewing` would be more appropriate & on L82

File: websites/W/watchmovieshd/presence.ts [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+		);
+		if (title) presenceData.details = title.textContent;
+		presenceData.largeImageKey = document
+			.querySelector("meta[property='og:image']")
+			.getAttribute("content");
+		if (iFrameData && !iFrameData.paused) {
+			[, presenceData.endTimestamp] = presence.getTimestamps(
+				iFrameData.currTime,
+				iFrameData.duration
+			);
```
Comment (EncryptedDev) [inline websites/W/watchmovieshd/presence.ts:70]:
```suggestion
			presenceData.endTimestamp = presence.getTimestamps(
				iFrameData.currTime,
				iFrameData.duration
			)[1];
```

File: websites/W/watchmovieshd/presence.ts [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+			if (episode) presenceData.state += ` - ${episode.textContent.trim()}`;
+		}
+		presenceData.largeImageKey = document
+			.querySelector("meta[property='og:image']")
+			.getAttribute("content");
+		if (!iFrameData?.paused) {
+			[, presenceData.endTimestamp] = presence.getTimestamps(
+				iFrameData.currTime,
+				iFrameData.duration
+			);
```
Comment (EncryptedDev) [inline websites/W/watchmovieshd/presence.ts:49]:
```suggestion
			presenceData.endTimestamp = presence.getTimestamps(
				iFrameData.currTime,
				iFrameData.duration
			)[1];
```

**Review by EncryptedDev — COMMENTED** [review]
Please resolve my comments and the the other two from contributors

File: websites/W/watchmovieshd/dist/metadata.json [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+		"name": "Toxicity",
+		"id": "794558195921846292"
+	},
+	"service": "watchmovieshd",
+	"description": {
+		"en": "Watch movies that get updated daily online in hd quality."
+	},
+	"url": "watchmovieshd.ru",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/iRop2fh.png",
```
Comment (theusaf) [inline websites/W/watchmovieshd/dist/metadata.json:13]: The website's logo is not a solid black square. Use the actual logo while ensuring that the resolution meets guidelines.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/W/watchmovieshd/dist/metadata.json (hunk 15 above)
Comment (Aarondevp) [reply websites/W/watchmovieshd/dist/metadata.json:13]: it isnt a black square that logo have a transparent background

Comment (theusaf) [reply websites/W/watchmovieshd/dist/metadata.json:13]: <details>
<summary>This is what I see</summary>
image
<p>No transparency to be seen, double check the image?</p>
</details>

Comment (RisingSunLight42) [reply websites/W/watchmovieshd/dist/metadata.json:13]: Can confirme it's a black square

Comment (Aarondevp) [reply websites/W/watchmovieshd/dist/metadata.json]: i think i just resolve it

File: websites/W/watchmovieshd/dist/metadata.json [hunk 16]
```diff
@@ -0,0 +1,30 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Toxicity",
+		"id": "794558195921846292"
+	},
+	"service": "watchmovieshd",
+	"description": {
+		"en": "Watch movies that get updated daily online in hd quality."
```
Comment (plskz) [inline websites/W/watchmovieshd/dist/metadata.json:9]:
```suggestion
		"en": "WatchMoviesHD is a free streaming service platform that allows its users to watch movies online for free."
```

File: websites/W/watchmovieshd/dist/metadata.json [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+	},
+	"url": "watchmovieshd.ru",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/9rVP8ZM.png",
```
Comment (plskz) [inline websites/W/watchmovieshd/dist/metadata.json:13]: logo should be `512x512` pixels.

**Review by plskz — CHANGES_REQUESTED** [review]
also, add **Screenshots** in your PR description.

File: websites/W/watchmovieshd/presence.ts (hunk 2 above)
Comment (theusaf) [reply websites/W/watchmovieshd/presence.ts:22]: Why was this resolved?

File: websites/W/watchmovieshd/presence.ts [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+		];
+	} else if (pathname === "/user/profile")
+		presenceData.details = "Viewing Profile";
+	else if (pathname === "/user/favourite")
+		presenceData.details = "checking Watchlist";
+	else {
+		const genre = document.querySelector<HTMLHeadingElement>("section.bl h1");
+		if (genre) {
+			presenceData.details = genre.textContent;
+			presenceData.smallImageKey = "search";
```
Comment (theusaf) [inline websites/W/watchmovieshd/presence.ts:87]: Use imgur. Same for [this comment](https://github.com/PreMiD/Presences/pull/6777/files#r967603658) after fixing [the logo](https://github.com/PreMiD/Presences/pull/6777/files#r971531061).

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/W/watchmovieshd/presence.ts (hunk 18 above)
Comment (Aarondevp) [reply websites/W/watchmovieshd/presence.ts:87]: where exactly

**Comment by EncryptedDev** [comment]
**Do not** mark my reviews as resolved if you haven't commented on them or merged the suggestion

**Comment by EncryptedDev** [comment]
Rebase

**Comment by EncryptedDev** [comment]
Sorry, please rebase your branch again
